lib/elix/responders/lists.ex

```diff
@@ -81,20 +65,31 @@ defmodule Elix.Responders.Lists do
   delete <item> from <name> - Deletes an item from a list by name or number
   """
   respond ~r/delete (.+) from (.+)/i, %Message{matches: %{1 => item_id, 2 => list_id}} = msg do
-    list_name = if Regex.match?(~r/\A\d+\Z/, list_id) do
+    list_name = parse_list_identifier(list_id)
+    item_name = parse_item_identifier(item_id, list_name)
+
+    Lists.delete_item(list_name, item_name)
+    reply(msg, render_list(list_name))
+  end
+
+  defp parse_list_identifier(list_id) do
+    if numeric_string?(list_id) do
```

Comment:
That's pretty dang clever! (Which, as you suggest, is reason enough not to use it 😉) After changing the `||`s to `or`s (I also forgot that guard clauses only allow the stricter versions of comparisons) it still doesn't quite do the trick, though I'm sure I could get it working with enough time.

Instead, I'm looking into possibly eliminating this function entirely and offloading the detection of numeric strings to the regex itself: https://github.com/indyelixir/elix/pull/6/files#r86683827